### ISSUE TITLE
[HOLD Expensify/issues/451223] [$250] Approver receives whisper when report is auto-approved on their behalf

### ISSUE THREAD

**Issue by garrettmknight** [Expensify/App#53708]
- Held on https://github.com/Expensify/Expensify/issues/451223 for the `SubmitReport` command to be refactored to be 1:1:1. Any other solution would be a hack.

If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** N/A
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @garrettmknight 
**Slack conversation** (hyperlinked to channel name): None, yet

## Action Performed:
1. Create a new workspace in NewDot
2. Invite a submitter to the workspace
3. Enable Workflows
4. In Workflows, enable Approvals
5. In Workflows, enable Delay Submission
6. Enable Rules
7. Enable auto-approval
8. Set 'Random report audit' to 0% to ensure auto-approval
9. As the submitter, create and submit an expense

## Expected Result:
The expense that was submitted should get auto-approved without notifying the approver.

## Actual Result:
The expense was auto-approved, but still notified the approver that they needed to approver via our whisper that's intended to only send when a report is approved and forwarded.

<img width="695" alt="Screenshot 2024-12-06 at 4 02 20 PM" src="https://github.com/user-attachments/assets/f4de7f9e-e2e1-46ba-bbb5-b63567840dc1">

## Workaround:
Yeah, but it's annoying.

## Platforms:

All

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021865064479090764280</li>
        <li>Upwork Job ID: 1865064479090764280</li>
        <li>Last Price Increase: 2024-12-06</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @tgolen</details>

**Comment by FitseTLT** [comment]
@garrettmknight This should be internal 👍

**Comment by eh2077** [comment]
Yeah, I also agreed that this should be internal and should be fixed from backend.

@zanyrenney Can you help add a `Hot Pick` tag for this issue? Thanks

**Comment by eh2077** [comment]
> Yeah, I also agreed that this should be internal and should be fixed from backend.
> 
> @zanyrenney Can you help add a `Hot Pick` tag for this issue? Thanks

@zanyrenney Friendly bump

**Comment by tgolen** [comment]
@Beamanator @cristipaval I wanted to get your thoughts on this one, since I saw you were recently discussing this exact whisper in https://github.com/Expensify/Auth/pull/13193/files#r1846155170.

The problem for this particular bug report is that:
1. The Auth command for `SubmitReport` is called first ([here](https://github.com/Expensify/Web-Expensify/blob/f091ae254cdb7584359651c3da5612c0328c62ba/lib/ReportAPI.php#L3551)) which sends the whisper
2. The Auth command for `ApproveReport` is called second ([here](https://github.com/Expensify/Web-Expensify/blob/f091ae254cdb7584359651c3da5612c0328c62ba/lib/ReportAPI.php#L3557)) which auto-approves the report

Thus, there is no way for `SubmitReport` to know if the report has been auto-approved or not and the whisper will always get sent.

I cannot think of a very good way of fixing this, and I'm wondering if you have any ideas. About the only thing I can think of is to add some kind of check before calling `SubmitReport` to see if the report _will_ get auto-approved, and if so, then pass a parameter down to `SubmitReport` to skip sending the whisper. This however feels like a chicken-and-egg situation because you can't know if it can be auto-approved until it is submitted.

**Comment by Beamanator** [comment]
Yeahhhh so I see your point and was thinking that could potentially work (check if the report _will_ get auto-approved in `SubmitReport`) - another solution which is definitely harder would be to make `SubmitReport` 1:1:1 - thennnn, in Auth we would know if the report is about to get auto-approved - and if not, we send the whisper to the submitter-to approver, if SO we send the whisper to the next approver

**Comment by tgolen** [comment]
Yeah, I briefly talked about this with Cristi last night and we had a few thoughts.

- We agreed 1:1:1 would be the best option (requires a lot of work)
- We could send the whisper from PHP (but this goes in the opposite direction of 1:1:1)
- We could try to make the `ApproveReport` command find and delete the whisper that was created during `SubmitReport` (too much of a hack)

**Comment by Beamanator** [comment]
Nice, yeah agreed on all points haha. I wonder if anyone already started working on making `SubmitReport` 1:1:1?

**Comment by tgolen** [comment]
OK, yes, it is being worked on in https://github.com/Expensify/Expensify/issues/451223. I am going to place this on HOLD until that is done, then we can come back and work on this.

If it is a problem in the meantime, I think the only thing we should consider is temporarily removing the whisper.

**Comment by tgolen** [comment]
# Monthly Update
- This is still on HOLD until #451223 is finished

### Next Steps
- Wwait for that issue to be finished

#### ETA
- None

**Comment by tgolen** [comment]
# Monthly Update
- Same

**Comment by tgolen** [comment]
# Monthly Update
- Same

**Comment by tgolen** [comment]
# Monthly Update
- Same

**Comment by tgolen** [comment]
# Month Update
- Same

**Comment by zanyrenney** [comment]
Assume same as above @tgolen ?

**Comment by tgolen** [comment]
Yep, still on HOLD for https://github.com/Expensify/Expensify/issues/451223

**Comment by tgolen** [comment]
Same

**Comment by tgolen** [comment]
Still on HOLD

**Comment by tgolen** [comment]
Still on HOLD

**Comment by tgolen** [comment]
I'm just going to close this for now. Reopen it if you'd like something to happen with it.
